flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/DataStream.java

```diff
@@ -654,6 +648,63 @@ public ExecutionConfig getExecutionConfig() {
 		return transform("Process", outputType, operator);
 	}
 
+	/**
+	 * Applies the given {@link KeyedProcessFunction} on the input stream, thereby creating a transformed output stream.
+	 *
+	 * <p>The function will be called for every element in the input streams and can produce zero
+	 * or more output elements.
+	 *
+	 * @param keyedProcessFunction The {@link ProcessFunction} that is called for each element in the stream.
+	 *
+	 * @param <K> The type of key in {@code KeyedProcessFunction}.
+	 *
+	 * @param <R> The type of elements emitted by the {@code KeyedProcessFunction}.
+	 *
+	 * @return The transformed {@link DataStream}.
+	 */
+	@PublicEvolving
+	public <K, R> SingleOutputStreamOperator<R> process(KeyedProcessFunction<K, T, R> keyedProcessFunction) {
```

Comment:
Does it make sense to add `process` with a `KeyedProcessFunction` on non keyed `DataStream`?